kview/src/test/java/dev/ikm/komet/kview/mvvm/login/LoginTest.java

```diff
@@ -0,0 +1,225 @@
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.MockedStatic;
+import org.testfx.api.FxToolkit;
+import org.testfx.framework.junit5.ApplicationTest;
+import org.testfx.util.WaitForAsyncUtils;
+
+import java.io.File;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+
+public class LoginTest extends ApplicationTest {
+
+    private Path tempDirectory;
+    private String originalUserHome;
+    private MockedStatic<EvtBusFactory> mockedStaticEvtBusFactory;
+
+    /**
+     * Creates the temporary directory and the users.ini file.
+     */
+    private void createUsersFile() {
+        try {
+            // Create temporary directory for user home
+            tempDirectory = Files.createTempDirectory("mockedHome");
+
+            // Store the original user.home property to restore it later
+            originalUserHome = System.getProperty("user.home");
+
+            // Set the system property for user.home to the temporary directory
+            System.setProperty("user.home", tempDirectory.toString());
+
+            // Create the "Solor" directory
+            Path solorDir = tempDirectory.resolve("Solor");
+            if (Files.notExists(solorDir)) {
+                Files.createDirectory(solorDir);
+            }
+
+            // Create and populate the users.ini file
+            Path usersFile = solorDir.resolve("users.ini");
+            try (BufferedWriter writer = Files.newBufferedWriter(usersFile)) {
+                writer.write("""
+                    # Define users and passwords
+                    [users]
+                    user.test = secret123, user
+                    """);
+            }
+
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to create temporary directory or users.ini file", e);
+        }
+    }
+
+    /**
+     * Cleans up the temporary directory and restores the original user.home system property.
+     */
+    private void cleanupFiles() throws IOException {
+        // Restore the original user.home system property
+        if (originalUserHome != null) {
+            System.setProperty("user.home", originalUserHome);
+        }
+
+        // Clean up the temporary directory
+        if (tempDirectory != null) {
+            try (Stream<Path> paths = Files.walk(tempDirectory)) {
+                paths.map(Path::toFile).forEach(File::delete);
+            }
+        }
+    }
+
+    @BeforeEach
+    void setup() throws Exception {
+        // Register the primary stage in TestFX
+        FxToolkit.registerPrimaryStage();
+
+        // Create the users.ini file in the temporary directory
+        createUsersFile();
```

Comment:
Question: if the file exists does it ignore the file?
Reasoning: Assumes the password can be changed manually and the file does not get overwritten.

Reply:
Correct! Soon we will make a commit related to the `LoginTest` in the case the file exists.